wallet-new/src/Cardano/Wallet/Client.hs

```diff
@@ -44,10 +45,13 @@ import qualified Pos.Core as Core
 --
 -- Other functions may be defined in terms of this 'WalletClient' -- see
 -- 'getWalletIndex' as a convenience helper for 'getWalletIndexPaged'.
+-- TODO(ks): I don't think that it's important to preserve paging as
+-- an important detail, we should remove paging and return the full set
+-- of results.
```

Comment:
The `WalletClient m` should contain the primitives of the API. We can easily implement something like `streamAddresses :: WalletClient m -> Producer m Address` that pages through on top if we need.